File: SecondYearCS/OOP/Civitas/src/civitas/SorpresaPagarCobrar.java
package civitas;

import java.util.ArrayList;

public class SorpresaPagarCobrar extends Sorpresa{
      
    private float valor;
    
  
    //Resto de sorpresasa
    SorpresaPagarCobrar(float valor, String texto){
        super(texto);
        this.valor = valor;
    }
  
    /**
     * @brief Se informa y se modifica el saldo del jugador actual con el valor de la sorpresa
     */
    @Override
    public void aplicarAJugador(int actual, ArrayList<Jugador> todos){
        if( this.jugadorCorrecto(actual, todos) ){
            informe(actual,todos);
            todos.get(actual).modificarSaldo(this.valor);
        }
    }


    @Override
    public String toString(){
       return super.toString() + " valor: " + valor;
    }

}

File: SecondYearCS/OOP/CivitasRB/lib/SorpresaPagarCobrar.rb
require_relative 'sorpresa.rb'
module CivitasRB
  class SorpresaPagarCobrar < Sorpresa
    attr_accessor :valor
    #Constructores por defecto y con parametro asdfas
    def initialize(valor,texto)
      super(texto)
      @valor = valor
    end

    #Modificamos el saldo del jugador según el valor indicado
    def aplicarAJugador(act,todos)
      if self.jugadorCorrecto(act, todos)
        self.informe(act,todos)
        todos.at(act).modificarSaldo(@valor)
      end
    end
    
    def to_s
      return super.to_s +  " valor: " + @valor +  " "
    end
    
  end
end